### ISSUE TITLE
HTML references using label-names

### ISSUE THREAD

**Issue by HansOlsson** [brucemiller/LaTeXML#895]
With "--splitnaming=labelrelative" the labels can be used to generate good names for the generated HTML-documents. 

I would want a similar **option** for the anchors and references inside the generated HTML - instead of the current use of auto-generated IDs, including in the generated side-bar and table-of-contents. 

The benefits are that the HTML-pages become more acceptable to users, the current S3.SS1 is just an obvious distraction for new users. Additionally the references become more stable, since sections are more likely to be added than renamed. Renaming can be handled if multiple labels generate multiple anchors.

As far as I can see it is possible with some modifications of Scan.pm and CrossRef.pm.

It might be that this feature exists in some other way, but I just cannot find it.

**Comment by brucemiller** [comment]
@fred-wang had a similar request in #668.  There's a bit of timing issue in the current scheme: id's are generated as the document is built, with ids typically be derived from their parents (you can customize the "s", "ss" prefix, btw by looking for NewCounter in bindings, but maybe awkward). In any case, the ids are created usually before any `\label` has been seen.

However, there may be a strategy after the document is completed (before or at the beginning of  Scan) to find all labels and replace the corresponding ids (and perhaps replace the beginning portion of children id's as well).  ... when somebody's got some spare time, that is!

**Comment by HansOlsson** [comment]
I see. That doesn't seem that difficult (ignoring children and some nodes that I am not familiar with) - and it seems I could get it to work in a fairly straightforward way.

But I did not add an option for it, etc.

**Comment by HansOlsson** [comment]
My code is in 
https://github.com/HansOlsson/LaTeXML/tree/UseLabel

I did not turn it into a pull-request, since I didn't add an option for it. (and didn't add a test-case.)

**Comment by brucemiller** [comment]
Wow! Impressed that you found your way through the sources! Unfortunately it can fail in corner cases. You change id's, but in rare cases there could already be links (idref) to those id's that would need to be changed.

The other issues are less important: it's probably not necessary to worry about children's id's; and this could be seen as a logical extension of `labelrelative` and so wouldn't need a distinct option.

But I'm afraid to do it Right needs a complete pass through the document changing both id's and idref's.

**Comment by brucemiller** [comment]
I took another look at your code, and got hit by an inspiration: Instead of changing the ID's within the document and trying to manage it there, use the nice bottle-neck code that generates a "fragment id" for the id's within the target document (leaving the XML document's ids unchanged) (See a3cf4d01).  This seems to work pretty well.  In fact, I wish I'd done this long ago; it's much nicer for LaTeXML's own site and manual!  Also, rather than introduce a new option, I've made this take effect whenever one of the label based page splits is used.

Thanks for the suggestion!
